### PR TITLE
Fix workflow and tagging strategy

### PR DESCRIPTION
**Pull request by llarsson**
Addresses #2

### CODE REVIEW

File: .github/workflows/build-and-deploy-to-ck8s.yaml [hunk 1]
```diff
@@ -7,11 +7,10 @@ on:
       - demo-*
 
 env:
-  DOMAIN: ckdemo.a1ck.io
-  APP_DOMAIN: ${{ github.ref_name }}.ckdemo.a1ck.io
+  DOMAIN: CHANGE_ME
```
Comment (cristiklein) [inline .github/workflows/build-and-deploy-to-ck8s.yaml:10]: Consider the more provocative version `I_DIDNT_READ_THE_README`. :smile:

File: .github/workflows/build-and-deploy-to-ck8s.yaml [hunk 2]
```diff
@@ -63,11 +61,15 @@ jobs:
       run: |-
         echo $DOCKER_PASSWORD \
           | docker login --username $DOCKER_USER --password-stdin harbor.$DOMAIN
+    
+    - name: Set container image tag
+      run: |-
+        echo "TAG=${{ github.ref_name }}-$(git describe --tags --always)" >> $GITHUB_ENV
```
Comment (cristiklein) [inline .github/workflows/build-and-deploy-to-ck8s.yaml]: I'd rather use [environment variables](https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables) as opposed to [contexts](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context) in shell snippets. Makes the whole thing more reusable across CI/CD solutions.

```suggestion
        echo "TAG=${GITHUB_REF_NAME}-$(git describe --tags --always)" >> $GITHUB_ENV
```

File: .github/workflows/build-and-deploy-to-ck8s.yaml [hunk 3]
```diff
@@ -7,11 +7,10 @@ on:
       - demo-*
 
 env:
-  DOMAIN: ckdemo.a1ck.io
-  APP_DOMAIN: ${{ github.ref_name }}.ckdemo.a1ck.io
+  DOMAIN: CHANGE_ME
+  APP_DOMAIN: ${{ github.ref_name }}.CHANGE_ME
   DOCKER_USER: 'robot$demo+ci-cd'
   REGISTRY_PROJECT: demo
```
Comment (cristiklein) [inline .github/workflows/build-and-deploy-to-ck8s.yaml:13]: I'm confused. Why not simply:

```suggestion
  REGISTRY_PROJECT: demo
  TAG: ${{ github.ref_name }}-${{ github.sha }}
```

Getting philosophical, but I'd expect either:

```
TAG=v1.0.1 # for a tag
TAG=branch_name-random_crap_like_job_id_or_sha # for a branch
```

File: .github/workflows/build-and-deploy-to-ck8s.yaml (hunk 1 above)
Comment (llarsson) [reply .github/workflows/build-and-deploy-to-ck8s.yaml:10]: No thanks, I don't think provocation is useful for achieving the desired outcome.

File: .github/workflows/build-and-deploy-to-ck8s.yaml (hunk 3 above)
Comment (llarsson) [reply .github/workflows/build-and-deploy-to-ck8s.yaml:13]: Great suggestion, have gone with this approach!

File: .github/workflows/build-and-deploy-to-ck8s.yaml (hunk 2 above)
Comment (llarsson) [reply .github/workflows/build-and-deploy-to-ck8s.yaml]: Have removed this line and gone with the other approach you suggested instead!

**Pull request merged**